### PR TITLE
[CLEAN] Remove OUI_ANTARES NON_ANTARES

### LINKED ISSUE #1144: Remove `OUI_ANTARES`, `NON_ANTARES`, use `bool` instead
143 matches for `OUI_ANTARES`, 100 matches for `NON_ANTARES`.

All of these variables are in **solver/optimisation** and **solver/simulation**.

### PR DESCRIPTION
**Pull request by payetvin**
close #1144

### CODE REVIEW

**Review by flomnes — CHANGES_REQUESTED** [review]
LGTM. Remove the def of OUI_ANTARES and NON_ANTARES

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 1]
```diff
@@ -200,10 +200,7 @@ bool OPT_AppelDuSimplexe(PROBLEME_HEBDO* problemeHebdo,
 
     Probleme.TypeDePricing = PRICING_STEEPEST_EDGE;
 
-    if (PremierPassage == NON_ANTARES)
-        Probleme.FaireDuScaling = NON_SPX;
-    if (PremierPassage == OUI_ANTARES)
-        Probleme.FaireDuScaling = OUI_SPX;
+    Probleme.FaireDuScaling = ( PremierPassage ? OUI_SPX : NON_SPX );
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp:203]: Why not make **FaireDuScaling** a boolean ?

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 2]
```diff
@@ -220,10 +217,7 @@ bool OPT_AppelDuSimplexe(PROBLEME_HEBDO* problemeHebdo,
     Probleme.CoutsReduits = ProblemeAResoudre->CoutsReduits;
 
 #ifndef NDEBUG
-    if (PremierPassage == OUI_ANTARES)
-        Probleme.AffichageDesTraces = NON_SPX;
-    else
-        Probleme.AffichageDesTraces = OUI_SPX;
+    Probleme.AffichageDesTraces = ( PremierPassage ? OUI_SPX : NON_SPX );
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp:220]: Why not make **AffichageDesTraces** a boolean ?

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 3]
```diff
@@ -267,7 +261,7 @@ bool OPT_AppelDuSimplexe(PROBLEME_HEBDO* problemeHebdo,
     optimizationStatistics->addSolveTime(solveTime);
 
     ProblemeAResoudre->ExistenceDUneSolution = Probleme.ExistenceDUneSolution;
-    if (ProblemeAResoudre->ExistenceDUneSolution != OUI_SPX && PremierPassage == OUI_ANTARES)
+    if (ProblemeAResoudre->ExistenceDUneSolution != OUI_SPX && PremierPassage)
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp:264]: why not make **ExistenceDUneSolution** a boolean ?

File: src/solver/optimisation/opt_construction_contraintes_couts_demarrage.cpp [hunk 4]
```diff
@@ -325,7 +325,7 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaireCoutsDeDemarrage(
         }
     }
 
-#if VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES != OUI_ANTARES
+#if !VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_construction_contraintes_couts_demarrage.cpp]: As the following block is never activated/compile, we should consider removing it

File: src/solver/optimisation/opt_gestion_des_bornes_couts_demarrage.cpp [hunk 5]
```diff
@@ -112,7 +112,7 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaireCoutsDeDemarrage(
                   = CorrespondanceVarNativesVarOptim
                       ->NumeroDeVariableDuNombreDeGroupesQuiTombentEnPanneDuPalierThermique[palier];
                 Xmin[var] = 0;
-#if VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES != OUI_ANTARES
+#if !VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_gestion_des_bornes_couts_demarrage.cpp]: Remove one of these 2 alternatives ?

File: src/solver/simulation/adequacy.cpp [hunk 6]
```diff
@@ -172,7 +172,7 @@ bool Adequacy::year(Progression::Task& progression,
             *pProblemesHebdo[numSpace], state.weekInTheYear, numSpace, hourInTheYear);
 
         // Reinit optimisation if needed
-        pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim ? OUI_ANTARES : NON_ANTARES;
+        pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim ? true : false;
```
Comment (guilpier-code) [inline src/solver/simulation/adequacy.cpp]:
```C++
pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim;
```
should be enough

File: src/solver/simulation/economy.cpp [hunk 7]
```diff
@@ -165,7 +165,7 @@ bool Economy::year(Progression::Task& progression,
           *pProblemesHebdo[numSpace], state.weekInTheYear, numSpace, hourInTheYear);
 
         // Reinit optimisation if needed
-        pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim ? OUI_ANTARES : NON_ANTARES;
+        pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim ? true : false;
```
Comment (guilpier-code) [inline src/solver/simulation/economy.cpp]:
```C++
pProblemesHebdo[numSpace]->ReinitOptimisation = reinitOptim;
```
should be enough

Comment (guilpier-code) [reply src/solver/simulation/economy.cpp]: Moreover, there is code duplication between **economy.cpp** and **adequacy.cpp**.
We should consider removing them.
For instance : 
```C++
catch (Data::UnfeasibleProblemError&)
{
    // need to clean next problemeHebdo
    reinitOptim = true;

    // Indicate failed week list (first week of the year is "week number one" for the
    // user but w=0 for the loop)
    failedWeekList.push_back(w + 1);

    // Define if simulation must be stopped
    if (Data::stopSimulation(study.parameters.include.unfeasibleProblemBehavior))
    {
          return false;
    }
}
```

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 8]
```diff
@@ -49,7 +49,7 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
 
     auto& parameters = study.parameters;
 
-    problem.Expansion = parameters.expansion ? OUI_ANTARES : NON_ANTARES;
+    problem.Expansion = parameters.expansion ? true : false;
```
Comment (guilpier-code) [inline src/solver/simulation/sim_calcul_economique.cpp]:
```C++
problem.Expansion = parameters.expansion;
```

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 9]
```diff
@@ -82,8 +82,8 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
 
     problem.WaterValueAccurate
       = (study.parameters.hydroPricing.hpMode == Antares::Data::HydroPricingMode::hpMILP)
-          ? OUI_ANTARES
-          : NON_ANTARES;
+          ? true
```
Comment (guilpier-code) [inline src/solver/simulation/sim_calcul_economique.cpp]: Same as above

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 10]
```diff
@@ -103,11 +103,11 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
 
     problem.OptimisationAvecCoutsDeDemarrage
       = (study.parameters.unitCommitment.ucMode == Antares::Data::UnitCommitmentMode::ucMILP)
-          ? OUI_ANTARES
-          : NON_ANTARES;
+          ? true
```
Comment (guilpier-code) [inline src/solver/simulation/sim_calcul_economique.cpp]: Same as above

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 11]
```diff
@@ -103,11 +103,11 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
+          : false;
 
     problem.OptimisationAuPasHebdomadaire
-      = (parameters.simplexOptimizationRange == Data::sorWeek) ? OUI_ANTARES : NON_ANTARES;
+      = (parameters.simplexOptimizationRange == Data::sorWeek) ? true : false;
```
Comment (guilpier-code) [inline src/solver/simulation/sim_calcul_economique.cpp]: etc.

**Review by guilpier-code — DISMISSED** [review]

File: src/solver/optimisation/opt_constants.h [hunk 12]
```diff
@@ -21,4 +21,4 @@ constexpr char EGOISTE = 2;
 constexpr char ANTARES_SIMPLEXE = 1;
 constexpr char ANTARES_PNE = 2;
 
-constexpr char VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES = OUI_ANTARES;
+constexpr bool VARIABLES_MMOINS_MOINS_BORNEES_DES_2_COTES = true;
```
Comment (flomnes) [inline src/solver/optimisation/opt_constants.h]: This decl becomes unused, please remove it

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 13]
```diff
@@ -139,47 +135,46 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
-                                                                                : NON_ANTARES);
+          = ((area.hydro.reservoirManagement && !area.hydro.useHeuristicTarget) ? true
+                                                                                : false);
         problem.CaracteristiquesHydrauliques[i]->TurbinageEntreBornes
           = ((area.hydro.reservoirManagement
               && (!area.hydro.useHeuristicTarget || area.hydro.useLeeway))
-               ? OUI_ANTARES
-               : NON_ANTARES);
+               ? true
+               : false);
```
Comment (flomnes) [inline src/solver/simulation/sim_calcul_economique.cpp]: Please change
```cpp
bool myDecision = someCondition ? true : false;
```
to
```c
bool myDecision = someCondition;
```

**Review by flomnes — CHANGES_REQUESTED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**